src/DataValidation.py

```diff
@@ -192,6 +195,109 @@ def checkItemsThatShouldBeRequired():
+            if "requires" not in location:
+                continue
+
+            # convert to json so we don't have to guess the data type
+            location_requires = json.dumps(location["requires"])
+
+            DataValidation._checkLocationRequiresForItemValueWithRegex(values_requested, location_requires)
+        # Second, check region requires for the presence of item name
+        for region_name in DataValidation.region_table:
+            region = DataValidation.region_table[region_name]
+
+            if "requires" not in region:
+                continue
+
+            # convert to json so we don't have to guess the data type
+            region_requires = json.dumps(region["requires"])
+
+            DataValidation._checkLocationRequiresForItemValueWithRegex(values_requested, region_requires)
+        # then if something is requested, we loop items
+        if values_requested:
+
+            # get all the available values with total count
+            for item in DataValidation.item_table:
+                # if the item is already progression, no need to check
+                if not item.get("progression") and not item.get("progression_skip_balancing"):
+                    continue
+
+                item_count = item.get('count', None)
+                if item_count is None: #check with none because 0 == false
+                    item_count = '1'
+
+                for key, count in item.get("value", {}).items():
+                    if not values_available.get(key.lower().strip()):
+                        values_available[key] = 0
+                    values_available[key] += int(count) * int(item_count)
+
+            # compare whats available vs requested
+            errors = []
+            for value, count in values_requested.items():
+                if values_available.get(value, 0) < count:
+                    errors.append(f"   '{value}': {values_available.get(value, 0)} out of the {count} {value} worth of progression items required can be found.")
+            if errors:
+                raise ValidationError("There are not enough progression items for the following values: \n" + "\n".join(errors))
+
+    @staticmethod
+    def preFillCheckIfEnoughItemsForValue(world: World, multiworld: MultiWorld):
+        from .Helpers import get_items_with_value, get_items_for_player
+        player = world.player
+        values_requested = {}
+
+        for region in multiworld.regions:
+            if region.player != player:
+                continue
+
+            Manualregion = DataValidation.region_table.get(region.name, {})
+            if "requires" in Manualregion and Manualregion["requires"]:
+                region_requires = json.dumps(Manualregion["requires"])
+
+                DataValidation._checkLocationRequiresForItemValueWithRegex(values_requested, region_requires)
+
+            for location in region.locations:
+                ManualLocation = world.location_name_to_location.get(location.name, {})

```

Comment:
ManualLocation is a class in Locations.py. I know that's not imported here, but would prefer to avoid the ambiguity. Variable names shouldn't be capitalized, anyways, since that's usually only for class names... so should adjust the region one above, too.

Reply:
i'll be honest idk why I used ManualLocation instead of just manualLocation, ill post a pr to fix it soon